p2p_wallet/Scenes/Main/History/History.Repository.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Created by Giang Long Tran on 15.04.2022.
+//
+
+import Foundation
+
+/// The repository that works with transactions.
+protocol HistoryTransactionRepository {
+    /// Fetches a list of signatures, that belongs to the `address`.
+    ///
+    /// - Parameters:
+    ///   - address: the account address
+    ///   - limit: the number of transactions that will be fetched.
+    ///   - before: the transaction signature, that indicates the offset of fetching.
+    /// - Returns: the list of `SignatureInfo`
+    func getSignatures(address: String, limit: Int, before: String?) async throws -> [SolanaSDK.SignatureInfo]
+    
+    /// Fetch all data of the transaction
+    ///
+    /// - Parameter signature: The transaction signature
+    /// - Returns: `TransactionInfo`, that can be parsed later.
+    func getTransaction(signature: String) async throws -> SolanaSDK.TransactionInfo
+}
+
+extension History {
+    typealias TransactionRepository = HistoryTransactionRepository
+
+    class SolanaTransactionRepository: HistoryTransactionRepository {
+        private let solanaSDK: SolanaSDK
```

Comment:
SolanaSDK must be injected using Resolver
`@Injected private var solanaSDK: SolanaSDK`

Reply:
+